### PR TITLE
Make adjustments to Section Leaders

### PR DESCRIPTION
**Pull request by KrisXV**
https://www.smogon.com/forums/threads/section-leader-updates-proposal.3687279/

### CODE REVIEW

File: server/user-groups.ts [hunk 1]
```diff
@@ -135,8 +135,14 @@ export abstract class Auth extends Map<ID, GroupSymbol | ''> {
 			targetSymbol = Auth.defaultSymbol();
 		}
 
-		const group = Auth.getGroup(symbol);
+		let group = Auth.getGroup(symbol);
 		if (group['root']) return true;
+		if (room?.settings.section && Users.globalAuth.sectionLeaders.has(user.id) &&
```
Comment (AnnikaCodes) [inline server/user-groups.ts:140]:
```suggestion
		if (
			room?.settings.section &&
```

**Review by AnnikaCodes — COMMENTED** [review]
This seems fine; `config.js` on sim3 will need to be updated.

**Comment by AnnikaCodes** [comment]
The Section Leader rank should also be excluded from `/auth`, since it would only show SLs who are not global staff (the `/sectionleaders` command shows all SLs).

**Comment by KrisXV** [comment]
> This seems fine; `config.js` on sim3 will need to be updated.

Staff room mod join will need to be adjusted as well just to note; I’ll address comments in a sec

**Comment by KrisXV** [comment]
Ok did @AnnikaCodes

File: server/user-groups.ts [hunk 2]
```diff
@@ -135,8 +135,14 @@ export abstract class Auth extends Map<ID, GroupSymbol | ''> {
...
 		}
 
-		const group = Auth.getGroup(symbol);
+		let group = Auth.getGroup(symbol);
 		if (group['root']) return true;
+		if (room?.settings.section &&
+			room.settings.section === Users.globalAuth.sectionLeaders.get(user.id) &&
+			// Global drivers who are SLs should get room mod powers too
+			Users.globalAuth.atLeast(user, SECTIONLEADER_SYMBOL)) {
+			group = Auth.getGroup('@');
```
Comment (AnnikaCodes) [inline server/user-groups.ts:144]:
```suggestion
		if (
			room?.settings.section &&
			room.settings.section === Users.globalAuth.sectionLeaders.get(user.id) &&
			// Global drivers who are SLs should get room mod powers too
			Users.globalAuth.atLeast(user, SECTIONLEADER_SYMBOL)
		) {
			group = Auth.getGroup('@');
```

**Comment by AnnikaCodes** [comment]
> > This seems fine; `config.js` on sim3 will need to be updated.
> 
> Staff room mod join will need to be adjusted as well just to note; I’ll address comments in a sec

The default modjoin setting on new servers for Staff room should probably be adjusted as well, then.

**Comment by AnnikaCodes** [comment]
> The default modjoin setting on new servers for Staff room should probably be adjusted as well, then.

(`rooms.ts` line 1178; I can't suggest for some reason)

**Comment by KrisXV** [comment]
> > > This seems fine; `config.js` on sim3 will need to be updated.
> > 
> > 
> > Staff room mod join will need to be adjusted as well just to note; I’ll address comments in a sec
> 
> The default modjoin setting on new servers for Staff room should probably be adjusted as well, then.

did, sorry i missed this comment fsr

**Review by AnnikaCodes — APPROVED** [review]
Looks good, thanks!

**Pull request merged**